versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseCommitHelper.java

```diff
@@ -569,37 +575,93 @@ && contentTypeForPayload((byte) op.payload()) == NAMESPACE) {
                   return ConflictResolution.IGNORE;
                 }
               }
-              MergeKeyBehavior mergeKeyBehavior = mergeBehaviorForKey.apply(key);
-              checkArgument(
-                  mergeKeyBehavior.getResolvedContent() == null
-                      && mergeKeyBehavior.getExpectedTargetContent() == null,
-                  "Functionality for MergeKeyBehavior.resolvedContent and MergeKeyBehavior.expectedTargetContent is not yet implemented.");
-              MergeBehavior mergeBehavior = mergeKeyBehavior.getMergeBehavior();
+
+              MergeBehavior mergeBehavior = mergeBehaviors.mergeBehavior(key);
               switch (mergeBehavior) {
-                case NORMAL:
-                  keyDetailsMap.put(key, keyDetails(mergeBehavior, ConflictType.UNRESOLVABLE));
-                  return ConflictResolution.IGNORE;
                 case FORCE:
-                  keyDetailsMap.put(key, keyDetails(mergeBehavior, ConflictType.NONE));
-                  return ConflictResolution.IGNORE;
                 case DROP:
-                  keyDetailsMap.put(key, keyDetails(mergeBehavior, ConflictType.NONE));
-                  return ConflictResolution.DROP;
+                  MergeKeyBehavior mergeKeyBe = mergeBehaviors.useKey(false, key);
+                  // Do not plain ignore (due to FORCE) or drop (DROP), when the caller provided an
+                  // expectedTargetContent.
+                  if (mergeKeyBe.getExpectedTargetContent() == null) {
+                    keyDetailsMap.put(key, keyDetails(mergeBehavior, ConflictType.NONE));
+                    return mergeBehavior == MergeBehavior.FORCE
+                        ? ConflictResolution.IGNORE
```

Comment:
nit: maybe rename `ConflictResolution.IGNORE` to `ADD` for the sake of clarity? On the caller side it is not obvious that `IGNORE` will cause the action to be added to the commit.

Reply:
Let's do in a separate PR